src/Component.php

```diff
@@ -241,6 +243,35 @@ private function handleFor( DOMNode $node, array $data ) {
 		}
 	}
 
+	private function appendHTML( DOMNode $parent, $source ) {
+		$tmpDoc = new DOMDocument();
+		$tmpDoc->loadHTML( $source );
+		foreach ( $tmpDoc->getElementsByTagName('body')->item(0)->childNodes as $node ) {
+			$node = $parent->ownerDocument->importNode( $node, true );
+			$parent->appendChild( $node );
+		}
+	}
+
+	private function handleRawHtml( DOMNode $node, array $data ) {
+		if ( $this->isTextNode( $node ) ) {
```

Comment:
This looks like you should do an `instanceof DOMElement` check instead. What if there is a third type that is neither text nor an element? The current approach must fail then.

Reply:
I guess you might be right. It might be worth addressing separately IMO, as similar check is used in other places too.